zcash_client_backend/src/data_api/error.rs

```diff
@@ -36,8 +36,11 @@ pub enum Error<DataSourceError, CommitmentTreeError, SelectionError, FeeError> {
     /// An error in transaction proposal construction
     Proposal(ProposalError),
 
-    /// The proposal was structurally valid, but spending shielded outputs of prior multi-step
-    /// transaction steps is not yet supported.
+    /// The proposal was structurally valid, but tries to do one of these unsupported things:
+    /// * spending a prior shielded output;
+    /// * paying to an output pool for which the corresponding feature is not enabled;
+    /// * paying to a TEX address if the "transparent-inputs" feature is not enabled;
+    /// * exceeding implementation limits.
```

Comment:
```suggestion
    /// * spend a prior shielded output;
    /// * pay to an output pool for which the corresponding feature is not enabled;
    /// * pay to a TEX address if the "transparent-inputs" feature is not enabled;
    /// * exceed implementation limits.
```
I don't understand what "exceeding implementation limits" means.

Reply:
There's a corner case where we return this if there are $\geq 2^{31}$ ephemeral outputs, which would otherwise cause an arithmetic overflow.

Reply:
This reinforces my belief that there should be at most one ephemeral transparent output of any given step.

Reply:
That error case is now merged with `DiversifierSpaceExhausted`.